server/src/main/java/org/apache/druid/server/coordinator/rules/BroadcastDistributionRule.java

```diff
@@ -49,7 +49,8 @@ public CoordinatorStats run(
       params.getDruidCluster().getAllServers().forEach(
           eachHolder -> {
             if (colocatedDataSources.stream()
-                                    .anyMatch(source -> eachHolder.getServer().getDataSource(source) != null)) {
+                                    .anyMatch(source -> eachHolder.getServer().getDataSource(source) != null
+                                                        && !eachHolder.isMaintenance())) {
```

Comment:
Should be outside of `anyMatch()`: `!eachHolder.isMaintenance() && colocatedDataSources.stream().anyMatch(...)`

Reply:
:white_check_mark: